lib/hubspot/company.rb

```diff
@@ -39,18 +39,35 @@ def all(opts={})
       end
 
       # Finds a list of companies by domain
-      # {http://developers.hubspot.com/docs/methods/companies/get_companies_by_domain}
+      # {https://developers.hubspot.com/docs/methods/companies/search_companies_by_domain}
       # @param domain [String] company domain to search by
+      # @param options [Hash] Possible options are:
+      #    limit [Integer] for pagination
+      #    properties [Array] list of company properties to recieve
+      #    offset_company_id [Integer] for pagination (should be company ID)
       # @return [Array] Array of Hubspot::Company records
-      def find_by_domain(domain)
-        path = GET_COMPANY_BY_DOMAIN_PATH
-        params = { domain: domain }
-        raise Hubspot::InvalidParams, 'expecting Integer parameter' unless domain.try(:is_a?, String)
+      def find_by_domain(domain, options = {})
+        raise Hubspot::InvalidParams, 'expecting String parameter' unless domain.try(:is_a?, String)
+
+        limit = options.fetch(:limit, 100)
+        properties = options.fetch(:properties, ["domain", "createdate", "name", "hs_lastmodifieddate"])
```

Comment:
The documentation for the old endpoint doesn't specify which properties will be returned as part of the query, and the documentation for the new endpoint specifies that _no properties_ will be returned by default. For backwards compatibility we should get the same properties so we don't break code that expects properties that were filled in from the old endpoint. Is this default properties list the same as what the old endpoint was returning?